web-console/future-openshift/contextual-help/contextual-help.md

```diff
@@ -0,0 +1,85 @@
+1. To provide contextual help on using **new features**
+2. To provide hints on a flow/ the **next steps** a user should be thinking about.
+
+#### New Features
+![hint pattern - new features](img/hint-new-features.png)
+
+#### Next Steps
+![hint pattern - next steps](img/hint-next-step.png)
+
+### Design
+#### General
+![hint pattern](img/hint-general.png)
+* Hint area is shown in context of where it applies
+  * This may be above the page title for cases where the hints apply to the entire page of the application.
+  * This may be below a specific tab or area on a page for cases where the hints only apply to a portion.
+* Hint area is highlighted in blue
+* Hint must have a title (h3 style)
+* Hint must have some body text (body-copy style)
+* There can optionally be external links within the body text
+* There can optionally be a primary action below the body text
+* Title, body text, and primary action are left aligned
+* For cases without a primary action, there can optionally be an X in the top right corner to dismiss the message
+
+#### Colors
+* Font: `pf-blue-600`
+* Background: `pf-blue-50`
+* Primary Action: Standard PatternFly button colors
+
+#### Spacing
+![hint pattern spacing](img/hint-spacing.png)
+* `18px` border between content and Background
+* `30px` space between the body text and primary button (when used)
+
+## Infopop Pattern
+The infopop pattern is a means of providing help on particular use cases in the UI. It is similar to Field Level Help, but rather than defining items in the UI, the infopop should instruct users how to use information/data that’s already shown in the UI to accomplish tasks which may not be obvious, or which may need to happen outside of the UI.
+
+### Potential Use Cases
+
+* Inform users about tasks that may need to be done outside of the UI.
+
+![infopop task outside of ui](img/infopop-outside-ui.png)
+
+* Provide guidance for non-obvious task
+![infopop for non-obvious task](img/infopop-non-obvious.png)
+
+### Design
+There are two aspects of the Infopop pattern: the prompt and the popover.
+
+#### Prompt
+* The prompt should be a question or statement that speaks to a specific customer use case, e.g. “Do you want to create a new tag?”
+* The prompt should be styled as link text preceded by the PatternFly help icon (pf-icon-help).
+* Clicking the prompt brings up a popover which points to the prompt.
+
+#### Popover
+* The popover should use PatternFly’s headless popover component.
+* If it is helpful to include a title, then the simple popover component may be used.
+* The content of the popover should be brief and instructive, informing users what information they need to accomplish the task without unnecessary background information.
+* A link to more comprehensive documentation can optionally be included.
+* The popover should not be the only source of data necessary to perform the task. In other words, an advanced user should be accomplish the task without needing to look at the popover.
```

Comment:
Another nit: missing "able to" in "...an advanced user should be able to accomplish..."